### PR TITLE
Homepage

### PR DESCRIPTION
**Pull request by xerxes-on**
Started working on the Homepage

### CODE REVIEW

File: src/api/home.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import client from '@/api/client.js'
+
+const trending = () => {
```
Comment (volhaliferovich) [inline src/api/home.js:3]: Please rename your api requests to include actions, like getTrandingBooks, getSuggestions, createProfile

File: src/assets/css/main.css [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    src: url('/src/assets/fonts/Poppins/Poppins-Bold.ttf') format('truetype');
+    font-weight: 700;
+    font-style: normal;
+}
+
+@font-face {
+    font-family: 'Poppins';
+    src: url('/src/assets/fonts/Poppins/Poppins-Italic.ttf') format('truetype');
+    font-weight: 400;
+    font-style: italic;
```
Comment (volhaliferovich) [inline src/assets/css/main.css:29]: You should have different font family names, depending on your font-weight. Here you should go with "Poppins" for 400, "Poppins SB" for 600, "Poppins B" for 700 and "Poppins I" for italic style. And in the code you will use it just like font-family: "Poppins SB", without the necessity to specify font-weight again.

File: src/components/HomeComponents/Home.vue [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    } finally {
+        /* empty */
+    }
+})
+// Main content cards
+const classes = ['row-span-2', 'col-span-2', 'col-span-2', '', '']
+</script>
+
+<template>
+    <ErrorPopup v-if="error" v-model="error" message="{{error_message}}" />
```
Comment (volhaliferovich) [inline src/components/HomeComponents/Home.vue:53]: If you want to pass a dynamic variable to a component like this error_message, you should use a colon instead, not curly braces.
```suggestion
    <ErrorPopup v-if="error" v-model="error" :message="error_message" />
```

Comment (volhaliferovich) [reply src/components/HomeComponents/Home.vue:53]: Also please use a camelCase for variable names (errorMessage). And by the way here you can use only one variable for error handling (not error and error_message). Just use a string to store your message and check if you should show your error popup using !!error_message

File: src/components/HomeComponents/Home.vue [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+                                </span>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+
+                <h2 class="text-xl font-bold mb-4 text-center pt-10">Suggestions</h2>
+                <div class="relative overflow-hidden w-full">
+                    <!-- Scrolling container -->
+                    <Suggestions class="" />
```
Comment (volhaliferovich) [inline src/components/HomeComponents/Home.vue:117]: What is this empty class for?

File: src/components/HomeComponents/Suggestions.vue [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import { VueperSlides, VueperSlide } from 'vueperslides'
+import 'vueperslides/dist/vueperslides.css'
+import { useHomeStore } from '@/stores/home.js'
+
+const homeStore = useHomeStore()
+</script>
+
+<template>
+    <vueper-slides :infinite="true" duration="2000" autoplay class="no-shadow h-full" :visible-slides="3" :gap="2" :bullets="false" :slide-ratio="1 / 4" fixed-height="300px">
+        <vueper-slide v-for="book in homeStore.suggestions" :key="book.id" :image="book.images" class="rounded-2xl" />
```
Comment (volhaliferovich) [inline src/components/HomeComponents/Suggestions.vue:11]: You can pass your suggestions in the parent component without the need to useHomeStore here as well. Just use a prop for that.

File: src/router/index.js [hunk 6]
```diff
@@ -1,21 +1,52 @@
...
+            name: 'home',
+            component: () => import('@/views/Home.vue'),
+        },
+        {
+            path: '/profile',
+            name: 'profile',
+            component: () => import('@/views/Profile.vue'),
+        },
+        {
+            path: '/logout',
```
Comment (volhaliferovich) [inline src/router/index.js:28]: Do we really need to have a separate route and component for logging out? I suppose it should be just a button or link somewhere in the header. What do you think?

File: src/views/Profile.vue [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+</script>
+
+<template>
+    <Layout>
+        <div class="bg-primary p-6">
+            <div class="max-w-6xl mx-auto">
+                <UserInfo />
+                <UserReading />
+                <UserStats />
+                <LogoutButton />
```
Comment (volhaliferovich) [inline src/views/Profile.vue:16]: I can barely see this button on the profile page. It only has a white color. Why don't we add some background?

**Review by volhaliferovich — CHANGES_REQUESTED** [review]

**Comment by volhaliferovich** [comment]
Please update this branch, so that it's up-to-date with the main one.

**Comment by volhaliferovich** [comment]
Also, remove the "components" word from the folders under the components folder. And have them like "login", "profile"

File: src/assets/css/main.css (hunk 2 above)
Comment (xerxes-on) [reply src/assets/css/main.css:29]: I think its working without naming it besacuse in tailwind-confid.js I configured different weights to diff names so I can use font-bold or font-italic #3

File: src/router/index.js (hunk 6 above)
Comment (xerxes-on) [reply src/router/index.js:28]: Can we do a function?

File: src/views/Profile.vue (hunk 7 above)
Comment (xerxes-on) [reply src/views/Profile.vue:16]: Its profile,  fixed in profile branch :)

File: src/assets/css/main.css [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+@tailwind utilities;
+@font-face {
+    font-family: 'Poppins';
+    src: url('/src/assets/fonts/Poppins/Poppins-Regular.ttf') format('truetype');
+    font-weight: 400;
+    font-style: normal;
+}
+
+@font-face {
+    font-family: 'Poppins';
```
Comment (volhaliferovich) [inline src/assets/css/main.css:12]: It's still the same as the above one.

File: src/components/Home/Home.vue [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+import { onMounted, ref, watch } from 'vue'
+import { useHomeStore } from '@/stores/home.js'
+import { useProfileStore } from '@/stores/profile.js'
+import homeApi from '@/api/home.js'
+import profileApi from '@/api/profile.js'
+import Svg from '@/components/common/Image.vue'
+import ErrorPopup from '@/components/common/ErrorPopup.vue'
+import Suggestions from '@/components/Home/Suggestions.vue'
+
+const error = ref(false)
```
Comment (volhaliferovich) [inline src/components/Home/Home.vue:11]: Please remove this one, it's unused

File: src/components/Home/Home.vue [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+const errorMessage = ref('')
+onMounted(async () => {
+    try {
+        const response = await homeApi.getTrendingBooks()
+        const suggestions = await homeApi.getSuggestedBooks()
+        const response_user = await profileApi.getProfileDetails()
+        if (response.status === 200) {
+            homeStore.trending = response.data
+        } else {
+            error.value = true
```
Comment (volhaliferovich) [inline src/components/Home/Home.vue:32]: and here, where you assign something to it

File: src/components/Home/Home.vue [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+                                    <i :class="['text-yellow', n <= book.rating / 2 ? 'fa-solid fa-star' : 'fa-regular fa-star']"></i>
+                                </span>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+
+                <h2 class="text-xl font-bold mb-4 text-center pt-10">Suggestions</h2>
+                <div class="relative overflow-hidden w-full">
+                    <Suggestions :homeStore="homeStore" />
```
Comment (volhaliferovich) [inline src/components/Home/Home.vue:115]: I wanted that you pass suggestions (homeStore.suggestions), not a homeStore itself

File: src/components/Home/Suggestions.vue [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+const props = defineProps({
+    homeStore: {
+        required: true,
+    },
+})
+</script>
+
+<template>
+    <vueper-slides :infinite="true" duration="2000" autoplay class="no-shadow h-full" :visible-slides="3" :gap="2" :bullets="false" :slide-ratio="1 / 4" fixed-height="300px">
+        <vueper-slide v-for="book in props.homeStore.suggestions" :key="book.id" :image="book.images" class="rounded-2xl" />
```
Comment (volhaliferovich) [inline src/components/Home/Suggestions.vue:14]: fix here as well

File: src/router/index.js (hunk 6 above)
Comment (volhaliferovich) [reply src/router/index.js:32]: You'll never see a special logout route in the apps. We don't need to count for this case, so, remove it. Just handle your LoginButton

**Review by volhaliferovich — CHANGES_REQUESTED** [review]

File: src/assets/css/main.css (hunk 8 above)
Comment (xerxes-on) [reply src/assets/css/main.css:12]: I configured in tailwind.config.js

Comment (volhaliferovich) [reply src/assets/css/main.css:12]: Then, why do you need to duplicate it? You set them in the output.css and main.css

File: src/components/Home/Home.vue [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+                                    <i :class="['text-yellow', n <= book.rating / 2 ? 'fa-solid fa-star' : 'fa-regular fa-star']"></i>
+                                </span>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+
+                <h2 class="text-xl font-bold mb-4 text-center pt-10">Suggestions</h2>
+                <div class="relative overflow-hidden w-full">
+                    <Suggestions :homeStore="homeStore.suggestions" />
```
Comment (volhaliferovich) [inline src/components/Home/Home.vue:111]: Change the name of the prop, it's not a homeStore now...

**Review by volhaliferovich — CHANGES_REQUESTED** [review]

**Review by volhaliferovich — APPROVED** [review]

**Pull request merged**